src/snowflake/snowpark/table.py

```diff
@@ -657,6 +657,9 @@ def drop_table(self) -> None:
         Note that subsequent operations such as :meth:`DataFrame.select`, :meth:`DataFrame.collect` on this ``Table`` instance and the derived DataFrame will raise errors because the underlying
         table in the Snowflake database no longer exists.
         """
-        self._session.sql(
-            f"drop table {self.table_name}"
-        )._internal_collect_with_tag_no_telemetry()
+        if hasattr(self._session._conn, "table_registry"):
+            self._session._conn.table_registry.drop_table(self.table_name)
+        else:
+            self._session.sql(
+                f"drop table {self.table_name}"
+            )._internal_collect_with_tag_no_telemetry()
```

Comment:
I'm in general a little concerned about if-else statement in non-mock module code from performance, code maintainability perspective.

for prpr let's do it, but could you log a JIRA to see how can we improve this in the PubPr? maybe we could introduce a mock_table module, or the in memory db-store could help